### DISCUSSION TITLE
Background colour of input boxes doesn't match colour of page

### DISCUSSION THREAD

**Discussion by tomyems** [UKHomeOffice/design-system#344] [Ideas]
The background colour of the pages in the Home Office design is light grey (#f1f1f1). See [Home Office design system colours](https://design.homeoffice.gov.uk/styles/colour) for an explanation.

The background to the text inputs, which are inherited from the [GOV.UK Design System](https://design-system.service.gov.uk/components/text-input/) are white. Is this a problem?

Also note that the radio buttons use the light grey throughout.

![image](https://user-images.githubusercontent.com/93599300/156806167-b12b3928-f9d3-444b-b810-728572e32d30.png)

![image](https://user-images.githubusercontent.com/93599300/156806661-71c72228-e46d-49bc-aa59-6557ce3f56fd.png)

**Comment by daniel-ac-martin** [comment]
I think this _might_ be an issue with govuk-frontend that is fixed in more recent versions .
i.e. I need to update it.

The latest version looks like it does _not_ set the background colour. In which case the style would be more like the radios above than the text input.

**Comment by tomyems** [comment]
I had a look at v4.1.0 of GOV.UK frontend, in the Home Office-styled GOV.UK prototype kit and got this:

![image](https://user-images.githubusercontent.com/93599300/176888035-ad8c30e6-1c8f-46ac-9618-248612100b25.png)

It looks like the background colour of the text input is being given by the 'user agent stylesheet' of the browser, rather than being specified. This makes sense when compared to the [latest stylesheet for text input](https://github.com/alphagov/govuk-frontend/blob/8456499d3d7b08953e2abb092b17f020e0f4c29b/src/govuk/components/input/_index.scss#L18) from GOV.UK frontend, which has a comment:

> // setting any background-color makes text invisible when changing colours to dark backgrounds in Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1335476)
> // as background-color and color need to always be set together, color should not be set either

Though the firefox bug is now fixed.

In the [latest GOV.UK frontend stylesheet for radios](https://github.com/alphagov/govuk-frontend/blob/8456499d3d7b08953e2abb092b17f020e0f4c29b/src/govuk/components/radios/_index.scss#L78), the 'background' is set to transparent. This is what leads to them appearing grey (with the page background) in the Home Office design system.

One possible way to change the background colour of the radios would be to set the `background-color` of `.govuk-radios__label:before` to white.

**Comment by willrichards** [comment]
This has come up a couple of times recently because we don't have any guidance in the design system. Can we add this as an action to the backlog to update the style? @guermanc please can I check the accessibility team would be happy with this?

**Reply by tomyems** [reply]
@willrichards - is this to make the colour of text boxes grey, or the colour of radios white?

**Reply by willrichards** [reply]
The intention is to make background colour of the radios white.
